ArduCopter/land_detector.pde

```diff
@@ -64,16 +65,27 @@ static void update_land_detector()
             // landed criteria met - increment the counter and check if we've triggered
             if( land_detector_count < ((float)LAND_DETECTOR_TRIGGER_SEC)*MAIN_LOOP_RATE) {
                 land_detector_count++;
+
             } else {
                 set_land_complete(true);
             }
+            // landed maybe criteria met - increment the counter
+            if (land_detector_maybe_count < 2*LAND_DETECTOR_MAYBE_TRIGGER_SEC*MAIN_LOOP_RATE) {
```

Comment:
Because otherwise one fail will cause the counter to drop below the decision threshold on line 88 and we will get no benefit from having the decrement when failed implementation. This way it will take 1 second to pass with continuous good passes and also 1 second to fail if data data continuously fails.